packages/loader/container-loader/src/test/serializedStateManager.spec.ts

```diff
@@ -0,0 +1,275 @@
+};
+
+describe("serializedStateManager", () => {
+	let seq: number;
+	let logger: ITelemetryLoggerExt;
+
+	function generateSavedOp(type: MessageType = MessageType.Operation): ISequencedDocumentMessage {
+		return {
+			clientId: "Some client ID",
+			minimumSequenceNumber: 0,
+			sequenceNumber: seq++,
+			type,
+		} as any as ISequencedDocumentMessage;
+	}
+
+	beforeEach(async () => {
+		seq = 1;
+		logger = createChildLogger({ namespace: "fluid:testSerializedStateManager" });
+	});
+
+	it("can't get pending local state when offline load disabled", async () => {
+		const storageAdapter = new MockStorageAdapter();
+		const serializedStateManager = new SerializedStateManager(
+			undefined,
+			logger,
+			storageAdapter,
+			false,
+		);
+
+		await assert.rejects(
+			async () =>
+				serializedStateManager.getPendingLocalStateCore(
+					{
+						notifyImminentClosure: false,
+					},
+					"clientId",
+					new MockRuntime(),
+					resolvedUrl,
+				),
+			(error: Error) =>
+				errorFn(error, "Can't get pending local state unless offline load is enabled"),
+			"container can get local state with offline load disabled",
+		);
+	});
+
+	it("can't get pending local state when there is no base snapshot", async () => {
+		const storageAdapter = new MockStorageAdapter();
+		const serializedStateManager = new SerializedStateManager(
+			undefined,
+			logger,
+			storageAdapter,
+			true,
+		);
+
+		await assert.rejects(
+			async () =>
+				serializedStateManager.getPendingLocalStateCore(
+					{
+						notifyImminentClosure: false,
+					},
+					"clientId",
+					new MockRuntime(),
+					resolvedUrl,
+				),
+			(error: Error) => errorFn(error, "no base data"),
+			"container can get local state with no base snapshot",
+		);
+	});
+
+	it("can get snapshot from previous local state", async () => {
+		const pendingLocalState: IPendingContainerState = {
+			baseSnapshot: { id: "fromPending", blobs: {}, trees: {} },
+			snapshotBlobs: {},
+			pendingRuntimeState: {},
+			savedOps: [],
+			url: "fluid",
+		};
+		const storageAdapter = new MockStorageAdapter();
+		const serializedStateManager = new SerializedStateManager(
+			pendingLocalState,
+			logger,
+			storageAdapter,
+			true,
+		);
+		const state = await serializedStateManager.getPendingLocalStateCore(
+			{ notifyImminentClosure: false },
+			"clientId",
+			new MockRuntime(),
+			resolvedUrl,
+		);
+		assert.strictEqual(JSON.parse(state).baseSnapshot.id, "fromPending");
+	});
+
+	it("can get pending local state after attach", async () => {
+		const storageAdapter = new MockStorageAdapter();
+		const serializedStateManager = new SerializedStateManager(
+			undefined,
+			logger,
+			storageAdapter,
+			true,
+		);
+		// equivalent to attach
+		serializedStateManager.setSnapshot({ tree: { trees: {}, blobs: {} }, blobs: {} });
+		for (let num = 0; num < 10; ++num) {
+			serializedStateManager.addSavedOp(generateSavedOp());
+		}
+		await serializedStateManager.getPendingLocalStateCore(
```

Comment:
Is the reason you don't assert after this call because you're relying on the asserts in getPendingLocalStateCore? Is that safe to do?